internal/export/distro/registrations.go

```diff
@@ -163,21 +164,41 @@ func (reg *registrationInfo) update(newReg contract.Registration) bool {
 	return true
 }
 
-func (reg registrationInfo) processEvent(event *models.Event) {
-	// Valid Event Filter, needed?
+func (reg registrationInfo) processMessage(msg msgTypes.MessageEnvelope) {
+	var err error
+	ctx := context.WithValue(context.Background(), clients.CorrelationHeader, msg.CorrelationID)
+	if msg.ContentType == clients.ContentTypeCBOR {
+		err = reg.handleCBOR(msg, ctx)
+	} else {
+		err = reg.handleJSON(msg, ctx)
```

Comment:
Opportunity to be more strict about expected inputs:
`else if msg.ContentType == clients.ContentTypeJSON {`
`. . . `
`else { // new error 'Unknown content type'`

Reply:
Modified to use switch. Default case handles unexpected `Content-type`